### PR TITLE
Bind

### PR DESCRIPTION
**Pull request by jmrodri**
Initial implementation of bind.

### CODE REVIEW

File: pkg/ansibleapp/bind.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	output, err := client.RunImage("bind", clusterConfig, instance.Spec, parameters)
+
+	if err != nil {
+		log.Error("Problem running image", err)
+		return nil, err
+	}
+
+	log.Info("{" + string(output) + "}")
+
+	log.Notice("Calling getPodName")
```
Comment (eriknelson) [inline pkg/ansibleapp/bind.go]: I'd change these to Debug level but keep them in so they can be turned on and off by us.

File: pkg/ansibleapp/bind.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+func getPodName(output []byte, log *logging.Logger) (string, error) {
+	r, err := regexp.Compile(`^pod[ \"]*(.*?)[ \"]*created`)
+	if err != nil {
+		return "", err
+	}
+
+	podname := r.FindStringSubmatch(string(output))
+
+	if log != nil {
+		log.Notice(fmt.Sprintf("%v", podname))
```
Comment (eriknelson) [inline pkg/ansibleapp/bind.go]: Debug here too? What do you think?

File: pkg/ansibleapp/bind_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
+package ansibleapp
+
+import (
+	"fmt"
+	ft "github.com/fusor/ansible-service-broker/pkg/fusortest"
+	"testing"
+)
+
+func TestDecodeOutput(t *testing.T) {
```
Comment (eriknelson) [inline pkg/ansibleapp/bind_test.go]: :tada:

File: pkg/ansibleapp/types.go [hunk 4]
```diff
@@ -79,6 +84,14 @@ type ServiceInstance struct {
 	Parameters *Parameters `json:"parameters"`
 }
 
+type BindInstance struct {
+	Id         uuid.UUID   `json:"id"`
+	ServiceId  uuid.UUID   `json:"service_id"`
+	Parameters *Parameters `json:"parameters"`
+	// v---  not sure if I need the following ---v
```
Comment (eriknelson) [inline pkg/ansibleapp/types.go]: I agree, I don't *think* the spec is needed here.

File: scripts/asbcli/asbcli [hunk 5]
```diff
@@ -110,7 +110,7 @@ class App(object):
         answers = self._get_answers(service)
         print 'Service configured! Requesting broker to provision...'
         res = self._req_provision(service['id'], answers)
-        print 'Provision returned status code: %s' % res.status_code
+        print 'Provision returned status code: %s, id [%s]' % (res.status_code, service['id'])
```
Comment (eriknelson) [inline scripts/asbcli/asbcli]: This is helpful, thanks :+1:

File: pkg/ansibleapp/bind.go (hunk 1 above)
Comment (jmrodri) [inline pkg/ansibleapp/bind.go]: Done

File: pkg/ansibleapp/bind.go (hunk 2 above)
Comment (jmrodri) [inline pkg/ansibleapp/bind.go]: Done

File: pkg/ansibleapp/bind_test.go (hunk 3 above)
Comment (jmrodri) [inline pkg/ansibleapp/bind_test.go]: :clap:

File: pkg/ansibleapp/types.go (hunk 4 above)
Comment (jmrodri) [inline pkg/ansibleapp/types.go]: removed

File: scripts/asbcli/asbcli (hunk 5 above)
Comment (jmrodri) [inline scripts/asbcli/asbcli]: Maybe, it's not the instance id so not sure how helpful it might be.

**Comment by eriknelson** [comment]
ACK

**Pull request merged**